src/Symfony/Component/Messenger/Tests/Adapter/PhpAmqp/AmqpReceiverTest.php

```diff
@@ -0,0 +1,110 @@
+use Symfony\Component\Messenger\Tests\Fixtures\DummyMessage;
+use Symfony\Component\Messenger\Transport\Serialization\Serializer;
+use Symfony\Component\Serializer as SerializerComponent;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+
+class AmqpReceiverTest extends TestCase
+{
+    public function testItYieldsTheDecodedMessageAndAcknowledgeIt()
+    {
+        $serializer = new Serializer(
+            new SerializerComponent\Serializer(array(new ObjectNormalizer()), array('json' => new JsonEncoder()))
+        );
+
+        $envelope = $this->createMock(\AMQPEnvelope::class);
+        $envelope->method('getBody')->willReturn('{"message": "Hi"}');
+        $envelope->method('getHeaders')->willReturn(array(
+            'type' => DummyMessage::class,
+        ));
+
+        $connection = $this->createMock(Connection::class);
+        $connection->method('waitAndGet')->willReturn($envelope);
+        $connection->expects($this->once())->method('ack')->with($envelope);
+
+        $receiver = new AmqpReceiver($serializer, $connection);
+        /** @var \Generator $generator */
+        $generator = $receiver->receive();
+
+        foreach ($generator as $message) {
+            $this->assertEquals(new DummyMessage('Hi'), $message);
+
+            // Stop the generator
+            $generator->send(AmqpReceiver::STOP);
+        }
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Messenger\Tests\Adapter\PhpAmqp\InterruptException
+     */
+    public function testItNonAcknowledgeTheMessageIfAnExceptionHappened()
+    {
+        $serializer = new Serializer(
+            new SerializerComponent\Serializer(array(new ObjectNormalizer()), array('json' => new JsonEncoder()))
+        );
+
+        $envelope = $this->createMock(\AMQPEnvelope::class);
+        $envelope->method('getBody')->willReturn('{"message": "Hi"}');
+        $envelope->method('getHeaders')->willReturn(array(
+            'type' => DummyMessage::class,
+        ));
+
+        $connection = $this->createMock(Connection::class);
+        $connection->method('waitAndGet')->willReturn($envelope);
+        $connection->expects($this->once())->method('nack')->with($envelope);
+
+        $receiver = new AmqpReceiver($serializer, $connection);
+
+        /** @var \Generator $generator */
+        $generator = $receiver->receive();
+
+        foreach ($generator as $message) {
+            $generator->throw(new InterruptException('Well...'));
+        }
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Messenger\Tests\Adapter\PhpAmqp\WillNeverWorkException
+     */
+    public function testItRejectsTheMessageIfTheExceptionIsARejectMessageException()
+    {
+        $serializer = new Serializer(
+            new SerializerComponent\Serializer(array(new ObjectNormalizer()), array('json' => new JsonEncoder()))
+        );
+
+        $envelope = $this->createMock(\AMQPEnvelope::class);
```

Comment:
I believe we use `getMockBuilder(class)->getMock()` for broader phpunit compat, dunno if it's encore required, but at least we don't have any `createMock()` occurrence in the codebase

Reply:
Changed all of these.